File: RcvViewModelLiveData/app/src/main/java/com/example/rcvviewmodellivedata/UserAdapter.java
package com.example.rcvviewmodellivedata;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.UserViewHolder> {
    private List<User> mListUser;

    public UserAdapter(List<User> mListUser) {

        this.mListUser = mListUser;
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_user,parent,false);
        return new UserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        User user = mListUser.get(position);
        if (user == null) {

            return;
        }
        holder.imgAvatar.setImageResource(user.getImageAvatar());
        holder.tvName.setText(user.getName());
        holder.tvDescription.setText(user.getDescription());

    }

    @Override
    public int getItemCount() {
        if (mListUser!=null){
            return mListUser.size();
        }
        return 0;
    }

    public class UserViewHolder extends RecyclerView.ViewHolder{
        private ImageView imgAvatar;
        private TextView tvName;
        private TextView tvDescription;

        public UserViewHolder(@NonNull View itemView) {
            super(itemView);
            imgAvatar = itemView.findViewById(R.id.img_avatar);
            tvName = itemView.findViewById(R.id.tv_name);
            tvDescription = itemView.findViewById(R.id.tv_description);
        }
    }
}

File: Android/ExampleRetrofitUser/app/src/main/java/com/example/exampleretrofituser/adapter/UserAdapter.kt
package com.example.exampleretrofituser.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.exampleretrofituser.databinding.UserItemBinding
import com.example.exampleretrofituser.model.UserModel

class UserAdapter(private val context: Context, val userList: List<UserModel>):
    RecyclerView.Adapter<UserAdapter.UserViewHolder>() {
    inner class UserViewHolder(val userBinding: UserItemBinding): RecyclerView.ViewHolder(userBinding.root) {
        fun bindingData(user: UserModel){
            userBinding.tvName.text = user.name
            userBinding.tvUserName.text = user.username
            userBinding.tvEmail.text = user.email
            userBinding.tvPhone.text = user.phone
            userBinding.tvWebsite.text = user.website
            userBinding.tvStreet.text = user.address.street
            userBinding.tvSuite.text = user.address.suite
            userBinding.tvCity.text = user.address.city
            userBinding.tvZipcode.text = user.address.zipcode
            userBinding.tvNameCompany.text = user.company.name
            userBinding.tvCatchPhrase.text = user.company.catchPhrase
            userBinding.tvBs.text = user.company.bs
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        return UserViewHolder(UserItemBinding.inflate(LayoutInflater.from(parent.context),parent,false))
    }

    override fun getItemCount(): Int {
        return userList.size
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        holder.bindingData(userList[position])
    }
}